### ISSUE TITLE
Suggestion: more config options for Spells and rituals not just true/false

### ISSUE THREAD

**Issue by SirAron111** [MysticMods/Roots#155]
Suggestion: more config options for Spells and rituals not just true/false

It is currently really hard to specify what spells are allowed to do and what not, as you can only turn them on or off with the true/false statements in the config.

For example the ritual wind wall wards of mobs from approaching the area this seems to affect all entities. If there was a range setting it would be adjustable for modpacks. Or a Black/with list for certain entities to be able to pass trough I mean it blocks even dragons not just arrows and zombies.

There are many spells that could need some config to adjust them or black/withlist systems for modpack makers to make other mods compatible or prevent certain effects from affecting certain things.

Or simply to adjust cool downs damage and range of things like spells and rituals.

**Comment by Davoleo** [comment]
This was already discussed, and spells will be customizable in another way ( iirc via CraftTweaker )

_Fun Fact: I had to revert my changes when I first implemented many kinds of different config entries for spells_

**Comment by SirAron111** [comment]
OK good to know as long as there will be a good documentation on how to adjust them via tweaker i can manage

**Comment by duely** [comment]
Yeah, basically each class that has hard-coded variables (ritual length, amplifiers, values, etc) will instead consult a table at first-execution. That table can then be modified via CraftTweaker before the game launches.

**Comment by duely** [comment]
As an update to this, the master branch already has this implemented for each of the spells and their various properties. I got a bit confused as to the potentials of the ZenScript system with regards to generics, so I've had to take a step back to rework some of the code. 

I'll be extending it to Rituals as well, and the system will look something like this:

```javascript
import mods.roots.Spells;
import mods.roots.Spell;

var growth_infusion = Spells.getSpell("spell_growth_infusion") as Spell;
growth_infusion.setCooldown(10);
growth_infusion.setInteger("tick_count", 5);
growth_infusion.setCost(0, "terra_moss", 0.1);
growth_infusion.setCost(1, "pereskia", 0.5);
```

This would result in the growth spell's costs being changed to 0.1 of Terra Moss and 0.5 of Perskia, meanwhile changing the number of times it applies `randomTick` to the block to 5, from the default of 1, adding an additional 4 calls *per channeled activation of the spell*, but changing the cooldown to 10.

For `Spells.getSpell("spell_harvest")`, you'd be able to `setInteger("radius_x", 10)` to adjust the X radius, the Y radius, the Z radius, etc. 

While cast types are implemented as properties, I don't intend for them to be modifiable as they are an intrinsic part of the spell. Cooldowns obviously can be modified, as well as costs (I might adjust the system slightly).

Worth noting is the fact that each of the `setXXX` functions will return the instance of the Spell, allowing for chained operations such as `Spells.getSpell("growth_infusion").setCooldown(10).setInteger("tick_count", 5);`. Likewise, the `spell_` and `ritual_` prefixes will be optional, or rather, the code will apply them if they don't exist when attempting to locate the spell. 

Hope this adds a little more information! Still a work in progress, but I'm hopeful it'll be one of the main features for customisation/configuration of 3.1.
